server/src/main/java/chat/services/AuthorizationService.java

```diff
@@ -1,18 +1,31 @@
 package chat.services;
 
+import chat.database.UsersBase;
 import chat.model.User;
+import org.apache.log4j.Logger;
 
 import java.io.BufferedReader;
 import java.io.IOException;
 import java.io.PrintWriter;
+import java.sql.SQLException;
 import java.util.ArrayList;
 
 public class AuthorizationService {
     private static final String ENTER_NUMBER = "Enter the number '1' if you want to loggin or '2' if you want to registration: ";
-
+    private static final Logger LOGGER = Logger.getLogger(AuthorizationService.class);
     private ArrayList<User> clientBase = new ArrayList<>();
+    private UsersBase usersBase;
+
+    public User authorization(PrintWriter writer, BufferedReader reader) throws IOException, SQLException, ClassNotFoundException {
+
+        usersBase = new UsersBase();
```

Comment:
Don't create this object each time when someone trying to authorize, create it once when creating  AuthorizationService instance

Reply:
++